src/core/core.tooltip.js

```diff
@@ -701,10 +701,10 @@ module.exports = function(Chart) {
 						// Inner square
 						ctx.fillStyle = mergeOpacity(vm.labelColors[i].backgroundColor, opacity);
 						ctx.fillRect(pt.x + 1, pt.y + 1, bodyFontSize - 2, bodyFontSize - 2);
-						var textColor = mergeOpacity(vm.labelTextColors[i], opacity);
-						ctx.fillStyle = textColor;
 					}
 
+					var textColor = mergeOpacity(vm.labelTextColors[i], opacity);
```

Comment:
Before introducing this bug, the fillStyle was set [before all the loops](/chartjs/Chart.js/blame/2258199c22c9369db629b312b1ece51a77adfc42/src/core/core.tooltip.js#L662), meaning that [`beforeBody`](/chartjs/Chart.js/blame/2258199c22c9369db629b312b1ece51a77adfc42/src/core/core.tooltip.js#L673) and [`bodyItem.before`](/chartjs/Chart.js/blame/2258199c22c9369db629b312b1ece51a77adfc42/src/core/core.tooltip.js#L680) was also supposed to be drawn using `bodyFontColor`. Not sure if that still work, there is no more `ctx.fillStyle` initialization?